synapse/storage/databases/main/search.py

```diff
@@ -727,20 +738,25 @@ def _to_postgres_options(options_dict):
     return "'%s'" % (",".join("%s=%s" % (k, v) for k, v in options_dict.items()),)
 
 
-def _parse_query(database_engine, search_term):
+def _parse_query_for_sqlite(search_term: str) -> str:
     """Takes a plain unicode string from the user and converts it into a form
-    that can be passed to database.
-    We use this so that we can add prefix matching, which isn't something
-    that is supported by default.
+    that can be passed to sqllite's matchinfo().
```

Comment:
This is currently a no-op, but we will probably want to add stuff here to handle certain cases e.g. https://github.com/matrix-org/synapse/issues/3024